nautobot/dcim/graphql/types.py

```diff
@@ -53,6 +54,11 @@ class Meta:
         filterset_class = DeviceFilterSet
         exclude = ["_name"]
 
+    dynamic_groups = graphene.List("nautobot.extras.graphql.types.DynamicGroupType")
+
+    def resolve_dynamic_groups(self, args):
+        return DynamicGroup.objects.get_for_object(Device.objects.get(pk=self.pk))
```

Comment:
Until we refactor this as a Mixin or base, any reason why to not change to in all places?

```suggestion
        return DynamicGroup.objects.get_for_object(self.Meta.model.objects.get(pk=self.pk))
```

Just to minimize copy-paste issues.

Reply:
Is this level of indirection needed or does `DynamicGroup.objects.get_for_object(self)` work too?

Reply:
> Is this level of indirection needed or does `DynamicGroup.objects.get_for_object(self)` work too?

Is `self` a `Device` object? or a shadowed one for GraphQL? I'm not familiar how the GraphQL types are built in relation to the normal models.

Reply:
It is not in fact needed, it works with `DynamicGroup.objects.get_for_object(self)`.